### DISCUSSION TITLE
Is 8 bar loop length possible?

### DISCUSSION THREAD

**Discussion by weicj** [bangcorrupt/hacktribe#132] [Ideas]
Many thanks to all you guys for this great effort to elevate Electribe2 to a whole new level! 

With the both versions' features combined, I think my Hacktribed E2S is really close to the perfect groovebox, besides some minor bugs. This only small gap, IMHO, might be the feature to get a 8-bar loop length (greater is better, but 8 bars is sufficient in most cases).
Loopop had a solution for 8 bar with the help of an app, but you would compromise half of 16 tracks and it is not very easy to use actually. 
For the sake of hardware layout restriction, I can imagine the constant lit LED indicator represent 1-4 bar as it is right now, and a flashing LED represent 5-8 bar. Press the indicator once to locate from 1-4, twice for 5-8. I don't know how complicated to implement this, but if it is possible, this generation of Electribe will be a crazy stuff.

**Comment by bangcorrupt** [comment]
I added this to the [FAQ](https://github.com/bangcorrupt/hacktribe/wiki/FAQ#8-bar-patterns).

> I haven't looked at longer patterns yet. It might be as simple as increasing the iterator and giving it more data to read, but I doubt it.

Thank you very much for your support! :smiley:

**Reply by weicj** [reply]
Cool, that feature is highly expected, can't wait to see if it works. 

And don't mention my tiny contribution to this project, you really deserve much more 😄.

**Reply by scottishcammy** [reply]
Hi, I see the FAQ is still the same, that this hasn't been developed. I bought the unit specifically to use as a drum machine, And stupidly believed ChatGPT telling me the 4-bar restriction had been lifted. Is this something that may be solved at some point?

Or am I just seeing an issue that I shouldn't worry about?

My ideal use case was to play pattern of around 8 bars in on the pads and then move the creative flow to synths.

**Reply by bangcorrupt** [reply]
Longer patterns would need the pattern file format to change.  I plan to do this if I can, but I think anything that makes file formats incompatible with the factory firmware should wait for Hacktribe 3.  

Conditional triggers and probability would mitigate pattern length issues, I'll probably work on that first.
